### DISCUSSION TITLE
Use opacity in color classes generated with CSS variables with dynamic themes?

### DISCUSSION THREAD

**Discussion by jasonlam-swatow** [unocss/unocss#1556] [Q&A] [answered]
Inspired by [this discussion](https://github.com/unocss/unocss/discussions/629#discussioncomment-2699165), I've been able to generate custom classes with dynamic themed CSS variables:

CSS implementing root CSS variables under different themes:

```css
:root[theme="tech-blue"] {
  --color-0: #158cf5;
  --color-1: #ef4c6c;
  // ...
}
```

In unocss.config.cjs:

```js
rules: [
  [/^text-(.*)$/, ([, c], { theme }) => {
    if (theme.colors[c]) {
      return {
        color: theme.colors[c], // generates `text-color-0` to `text-color-10`
      }
    }
  }],
],
theme: {
  colors: (cnt => {
    const palette = {}
    for (let i = 0; i <= cnt; i++) {
      palette[`color-${i}`] = `var(--color-${i})`
    }
    return palette
  })(10) // 10 colors in each theme
}
```

In .vue template, I can write custom classes like `text-color-2`, and it works like a charm at build-time:

```html
<p class="text-color-2 bg-color-3">Test</p>
```

**But I can't find a way to use opacity in this kind of custom classes, like `text-color-3/50`.** Is there a way to configure, or do I have to write more RegExp to make it possible manually?

**Comment by zyyv** [comment, marked as answer]
Try to this: 

```ts
theme: {
  colors: (cnt => {
    const palette = {}
    for (let i = 0; i <= cnt; i++) {
      palette[`color-${i}`] = `rgba(var(--color-${i}),%alpha)`
    }
    return palette
  })(10) // 10 colors in each theme
}
```

**Reply by jasonlam-swatow** [reply]
Doesn't work. Below is what it produces:

<img width="1099" alt="image" src="https://user-images.githubusercontent.com/8470477/188579735-ab880aeb-3c55-4b00-81c2-702ff06a8527.png">

<img width="847" alt="image" src="https://user-images.githubusercontent.com/8470477/188579705-c458e95a-bfa3-43ba-aa31-8678750f1c31.png">

**Reply by zyyv** [reply]
You should rename you colors key, `-` may cause unocss to parse incorrectly.

[Playground](https://uno.antfu.me/play/?html=DwEwlgbgBAxgNgQwM5ILwCIBGBzAtDAVyQBcB7AWyh3yLPIC4BWABnQD4AoKKD4AenAQ2QA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmANHGFKgM6owCCMMUwARgK4zDoCeZF1tAJIBjAlT6UaMAKpYIcAL5x0UCCDgByNnOFUqGgFAHUAD0iwUGAIZsANvDSYc%2BQsCIAKBAbhwYAC1QQVAAuRG8fOFFbaCpQrwiI4TYqGDVQjSgiFit3ADcrKHcAWiKo6FLk1JAAShIAUitbMD8rao0ScJ8FDq6e8klaWLgAbU7%2BgRk5d1qx-ikGJlYOLm5pvp85oVEsKk8xnyphRpC4AEYAOgAmdcTkLHS-RjBYgHoX6hBzqj8X9rGFGY%2BAC6HQBBiAA&options=N4XyA)

**Reply by jasonlam-swatow** [reply]
It works! Thanks.

**Reply by jasonlam-swatow** [reply]
As for base format color CSS variables, it's quite inconvenient to use elsewhere. Is there any chance unocss would support hex/rgba in this scenario? 🙏🏻 @chris-zhu
